backend.go

```diff
@@ -305,3 +307,24 @@ type VersionedBackend interface {
 	// nil page identically to a zero page.
 	ListBucketVersions(bucketName string, prefix *Prefix, page *ListBucketVersionsPage) (*ListBucketVersionsResult, error)
 }
+
+func MergeMetadata(db Backend, bucketName string, objectName string, meta map[string]string) error {
```

Comment:
Placed this function here, as I had the felling it is probably most logically placed as a helper function for each backend implementation. As the `meta map[string]string` is already a reference, we are just passing a pointer to manipulate the `meta` `map`. Still we need to retrieve a potential existing item.